tests/tests_pytorch/strategies/test_sharded_strategy.py

```diff
@@ -18,6 +20,37 @@
     from fairscale.optim import OSS
 
 
+class ModelWithAdamOptimizer(BoringModel):
+    def configure_optimizers(self):
+        optimizer = torch.optim.Adam(self.layer.parameters(), lr=0.1)
+        return optimizer
+
+
+class CheckModelRestore(ModelWithAdamOptimizer):
+    def __init__(self, old_model_state_dict, old_optimizer_states):
+        super().__init__()
+        self.old_model_state_dict = old_model_state_dict
+        self.old_optimizer_states = old_optimizer_states
+
+    def on_train_start(self):
+        assert all(
+            self._is_equal(actual, expected) for actual, expected in zip(self.state_dict(), self.old_model_state_dict)
+        )
+
+        for optimizer, state in zip(self.trainer.optimizers, self.old_optimizer_states):
+            optimizer.consolidate_state_dict()
+            self._is_equal(optimizer.state_dict(), state)
+
+    def _is_equal(self, a, b):
+        if isinstance(a, torch.Tensor):
+            return torch.all(torch.eq(a, b))
```

Comment:
```suggestion
            return torch.allclose(a, b)
```

otherwise this might fail due to minor numeric inaccuracies. `allclose` == `eq` for integer tensors and `allclose` == `a - b < eps` for floating point :)